### PR TITLE
Update hub image dependencies and RELEASE.md regarding dependencies

### PR DESCRIPTION
**Pull request by consideRatio**
Making a z2jh release is sometimes motivated by bumping dependencies etc. In this PR I add an overview about the relevant dependencies within RELEASE.md and provide lots of relevant links from the hub image's requirements.txt file to facilitate version bumping.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -10,6 +10,48 @@ This issue will be used to coordinate the next release of the Helm chart
...
 
+## Look through dependencies
+
+The JupyterHub Helm chart relies on many dependent projects, and when we make a
+release it is good to be updated about their status and what version we decide
+to couple the Helm chart release with. Below are the more important depdencies.
+Put a check on those that reach a state good enough for a z2jh release to be
+cut.
+
+### Depdendent Python packages
```
Comment (manics) [inline RELEASE.md]:
```suggestion
### Dependent Python packages
```

File: RELEASE.md [hunk 2]
```diff
@@ -10,6 +10,48 @@ This issue will be used to coordinate the next release of the Helm chart
...
+A more complete list is available in the
+[images/hub/requirements.txt](images/hub/requirements.txt), but here are some of
+the big ones.
+
+- [ ] [jupyterhub](https://github.com/jupyterhub/jupyterhub)
+- [ ] [kubespawner](https://github.com/jupyterhub/kubespawner)
+  - [ ] [kubernetes-client/python](https://github.com/kubernetes-client/python)
+- [ ] [oauthenticator](https://github.com/jupyterhub/oauthenticator)
+
+### Depdendent docker images
```
Comment (manics) [inline RELEASE.md]:
```suggestion
### Dependent docker images
```

File: RELEASE.md [hunk 3]
```diff
@@ -10,6 +10,48 @@ This issue will be used to coordinate the next release of the Helm chart
...
+These images version/tags are set in [values.yaml](jupyterhub/values.yaml).
+
+- [ ] [configurable-http-proxy](https://github.com/jupyterhub/configurable-http-proxy)
+  - [Available image tags](https://hub.docker.com/r/jupyterhub/configurable-http-proxy/tags)
+  - values.yaml entry: proxy.chp.image
```
Comment (manics) [inline RELEASE.md]:
```suggestion
  - values.yaml entry: `proxy.chp.image`
```

File: RELEASE.md [hunk 4]
```diff
@@ -10,6 +10,48 @@ This issue will be used to coordinate the next release of the Helm chart
...
+- [ ] [kubernetes/ingress-nginx](https://github.com/kubernetes/ingress-nginx)
+  - [Available image tags](https://quay.io/repository/kubernetes-ingress-controller/nginx-ingress-controller?tab=tags)
+  - values.yaml entry: proxy.nginx.image
```
Comment (manics) [inline RELEASE.md]:
```suggestion
  - values.yaml entry: `proxy.nginx.image`
```

File: RELEASE.md [hunk 5]
```diff
@@ -10,6 +10,48 @@ This issue will be used to coordinate the next release of the Helm chart
...
+- [ ] [kube-scheduler](https://github.com/kubernetes/kube-scheduler)
+  - [Available image tags](https://gcr.io/google_containers/kube-scheduler-amd64)
+  - values.yaml entry: scheduling.userScheduler.image
```
Comment (manics) [inline RELEASE.md]:
```suggestion
  - values.yaml entry: `scheduling.userScheduler.image`
```

**Comment by meeseeksmachine** [comment]
This pull request has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/binder-jupyterhub-activity-round-up-week-1/2704/1

**Pull request merged**